packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/event_client/string_database/dart_string_database.web.dart

```diff
@@ -0,0 +1,112 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'dart:async';
+
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/event_client/string_database/index_db/in_memory_string_db.dart';
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/event_client/string_database/index_db/indexed_db_adapter.dart';
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/event_client/string_database/string_database.dart';
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/event_client/string_database/string_database_element.dart';
+import 'package:async/async.dart';
+import 'package:aws_common/aws_common.dart';
+import 'package:drift/drift.dart';
+import 'package:meta/meta.dart';
+
+/// {@macro amplify_analytics_pinpoint_dart.dart_string_database}
+class DartStringDatabase
+    with AWSDebuggable, AWSLoggerMixin
+    implements StringDatabase, Closeable {
+  /// {@macro amplify_analytics_pinpoint_dart.index_db_string_db}
+  // ignore: avoid_unused_constructor_parameters
+  DartStringDatabase(QueryExecutor? driftQueryExecutor);
+
+  final _initMemo = AsyncMemoizer<void>();
+
+  /// {@macro amplify_analytics_pinpoint_dart.index_db_string_db}
+  Future<void> _init() async {
+    await _initMemo.runOnce(
+      () async {
+        final isIndexedDBSupported =
+            await IndexedDbAdapter.checkIsIndexedDBSupported();
+
+        if (isIndexedDBSupported) {
+          _database = IndexedDbAdapter();
+        } else {
+          logger.warn(
+            'IndexedDB is not available. '
+            'Falling back to in-memory storage.',
+          );
+          _database = InMemoryStringDb();
+        }
+      },
+    );
+  }
+
+  late StringDatabase _database;
+
+  @override
+  String get runtimeTypeName => 'DartStringDatabase';
+
+  @override
+  Future<void> addString(String string) async {
+    await _init();
+
+    try {
+      _database.addString(string);
+    } on Exception catch (e) {
+      _handleException(e);
+    }
+  }
+
+  @override
+  Future<void> deleteStrings(Iterable<int> idsToDelete) async {
+    await _init();
+
+    try {
+      _database.deleteStrings(idsToDelete);
+    } on Exception catch (e) {
+      _handleException(e);
+    }
+  }
+
+  @override
+  Future<Iterable<StringDatabaseElement>> getStrings(int maxToGet) async {
+    await _init();
+
+    try {
+      return _database.getStrings(maxToGet);
+    } on Exception catch (e) {
+      _handleException(e);
+      return [];
```

Comment:
Shouldn't we throw this exception here instead of swallowing it? This seems like a critical piece that analytics functionality depends on; if there is an exception here, it seems that we did something wrong in the library in the first place.

Reply:
Indeed, per Jordan's comments above rewrote this logic to just throw exception.